richtextfx/src/main/java/org/fxmisc/richtext/TextFlowExt.java

```diff
@@ -141,23 +141,36 @@ CharacterHit hitLine(double x, int lineIndex) {
     CharacterHit hit(double x, double y) {
         HitInfo hit = textLayout().getHitInfo((float) x, (float) y);
         int charIdx = hit.getCharIndex();
+        boolean leading = hit.isLeading();
 
         int lineIdx = getLineIndex((float) y);
         if(lineIdx >= getLineCount()) {
             return CharacterHit.insertionAt(getCharCount());
         }
 
-        TextLine line = getLines()[lineIdx];
+        TextLine[] lines = getLines();
+        TextLine line = lines[lineIdx];
```

Comment:
Did you break this up so that it's easier to read? Or is this "noise" (as some might call it) unintentional?

Reply:
No. The `lines` variable is used in a if statement to check whether this is a wrapped paragraph 😉

Reply:
Ah... I see now. Sorry, I misread it.